python/guide/packages/gluon/hybridize.md

```diff
@@ -199,3 +199,81 @@ net(x)
+
+## Key differences and limitations of hybridization
+
+The difference between a purely imperative `Block` and hybridizable `HybridBlock` can superficially appear to be simply the injection of the `F` function space (resolving to [`mx.nd`](http://beta.mxnet.io/api/ndarray/index.html) or [`mx.sym`](http://beta.mxnet.io/api/symbol/index.html)) in the forward function that is renamed from `forward` to `hybrid_forward`. However there are some limitations that apply when using hybrid blocks. In the following section we will review the main differences, giving example of code snippets that generate errors when such blocks get hybridized.
+
+### Indexing
+
+When trying to access specific elements in a tensor like this:
+
+```python
+def hybrid_forward(self, F, x):
+    return x[0,0]
+```
+
+Would generate the following error:
+
+`TypeError: Symbol only support integer index to fetch i-th output`
+
+There are however several operators that can help you with array manipulations like: [`F.split`](http://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.split.html#mxnet.ndarray.split), [`F.slice`](http://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.slice.html#mxnet.ndarray.slice), [`F.take`](http://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.take.html),[`F.pick`](http://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.pick.html), [`F.where`](http://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.where.html), [`F.reshape`](http://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.reshape.html) or [`F.reshape_like`](http://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.reshape_like.html).
+
+### Data Type
+
+Sometimes one can be tempted to use conditionnal logic on the type of the input tensors however the following block:
+
+```python
+def hybrid_forward(self, F, x):
+    if x.dtype =='float16':
+        return x
+    return x*2
+```
+
+Would generate a `AttributeError: 'Symbol' object has no attribute 'dtype'`
+
+You cannot use the `dtype` of the symbol at runtime as symbols only describe operations and not the underlying data they operate on. One workaround is to pass the type as a constructor argument of your network and hence build the appropriate compute graph for each situation.
```

Comment:
This is because you cannot use the data type of a symbol at runtime. Symbols only describe operations and .... 

One work around is to pass type as a constructor argument (please include the code snippet below)

Reply:
I don't want to overload the tutorial, I think this enough details like that.